openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp

```diff
@@ -455,6 +464,90 @@ boost::optional<IdfObject> ForwardTranslator::translateAirLoopHVACUnitarySystem(
+  // Note it will never be user specified at this time, because OS does not have the UnitarySystemPerformanceMultispeed object
+  // The plan as of this writting is to eventually support UnitarySystemPerformanceMultispeed and if user specified, then
+  // don't do this automatic generation
+  
+  if( (coolingCoil && (coolingCoil->iddObjectType() == model::CoilCoolingDXMultiSpeed::iddObjectType())) || 
+      (heatingCoil && (heatingCoil->iddObjectType() == model::CoilHeatingDXMultiSpeed::iddObjectType())) ) {
+
+    IdfObject _unitarySystemPerformance(openstudio::IddObjectType::UnitarySystemPerformance_Multispeed);
+    m_idfObjects.push_back(_unitarySystemPerformance);
+    _unitarySystemPerformance.setName(unitarySystem.nameString() + " Unitary System Performance");
+    
+    boost::optional<model::CoilHeatingDXMultiSpeed> multispeedHeating;
+    boost::optional<model::CoilCoolingDXMultiSpeed> multispeedCooling;
+
+    int maxStages = 0;
+
+    if( heatingCoil ) {
+      multispeedHeating = heatingCoil->optionalCast<model::CoilHeatingDXMultiSpeed>();
+    }
+
+    if( coolingCoil ) {
+      multispeedCooling = coolingCoil->optionalCast<model::CoilCoolingDXMultiSpeed>();
+    }
+
+    std::vector<model::CoilHeatingDXMultiSpeedStageData> heatingStages;
+    std::vector<model::CoilCoolingDXMultiSpeedStageData> coolingStages;
+
+    if( multispeedHeating ) {
+      heatingStages = multispeedHeating->stages();
+      _unitarySystemPerformance.setInt(UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforHeating,heatingStages.size());
+      maxStages = heatingStages.size();
+    } else {
+      _unitarySystemPerformance.setInt(UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforHeating,1);
+    }
+
+    if( multispeedCooling ) {
+      coolingStages = multispeedCooling->stages();
+      _unitarySystemPerformance.setInt(UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforCooling,coolingStages.size());
+      int stages = coolingStages.size();
+      if( stages > maxStages ) {
+        maxStages = stages;
+      }
+    } else {
+      _unitarySystemPerformance.setInt(UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforCooling,1);
+    }
+
+    _unitarySystemPerformance.setString(UnitarySystemPerformance_MultispeedFields::SingleModeOperation,"Yes"); 
+
+    auto heatingFlow = modelObject.supplyAirFlowRateDuringHeatingOperation();
+    auto coolingFlow = modelObject.supplyAirFlowRateDuringCoolingOperation();
+
+    for( int i = 0; i < maxStages; ++i ) {
+      auto extensible = _unitarySystemPerformance.pushExtensibleGroup();
+
+      if( i < heatingStages.size() ) {
+        auto heatingStage = heatingStages[i];
+        auto stageFlow = heatingStage.ratedAirFlowRate();
+        if( stageFlow && heatingFlow ) {
+          extensible.setDouble(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio,stageFlow.get() / heatingFlow.get());
```

Comment:
@kbenne This needs to be a ratio, not flow rate. So probably `heatingFlow.get() / stageFlow.get()` or something like that.